Assets/Scripts/Framework/Gameplay/LevelTracker.cs

```diff
@@ -0,0 +1,200 @@
+using System.Collections.Generic;
+using GameKit.Utilities;
+using FishNet.Object;
+using UnityEngine;
+using System;
+
+using DerailedDeliveries.Framework.DamageRepairManagement.Damageables;
+using DerailedDeliveries.Framework.Gameplay.Level;
+using DerailedDeliveries.Framework.Utils;
+using System.Linq;
+
+namespace DerailedDeliveries.Framework.Gameplay
+{
+    /// <summary>
+    /// A class that generates the objectives and tracks the Level's progress.
+    /// </summary>
+    public class LevelTracker : NetworkAbstractSingleton<LevelTracker>
+    {
+        [SerializeField]
+        private Levels _levels;
+
+        [SerializeField]
+        private int _succesfullDeliveryBonus = 5;
+
+        [SerializeField]
+        private int _incorrectDeliveryPenalty = 5;
+
+        [SerializeField]
+        private int _totalScore;
+
+        [SerializeField]
+        private int _currentScore;
+
+        [SerializeField]
+        private GameObject _packagePrefab;
+
+        [SerializeField]
+        private TrainStation[] _allStations;
+
+        /// <summary>
+        /// An action that broadcasts when a package is delivered and the amount of points that is added.
+        /// </summary>
+        public Action<int> OnPackageDelivered;
+
+        private static readonly char[] CHARACTERS = "QWERTYUIOPASDFGHJKLZXCVBNM".ToCharArray();
+
+        /// <summary>
+        /// <inheritdoc/>
+        /// </summary>
+        public override void OnStartServer()
+        {
+            base.OnStartServer();
+
+            SelectLevelToLoad(0);
+        }
+
+        /// <summary>
+        /// A function that is used to load a new level.
+        /// </summary>
+        /// <param name="index">The level ID.</param>
+        [Server]
+        private void SelectLevelToLoad(int index)
+        {
+            _currentScore = 0;
+
+            StationLevelData[] levelData = _levels.levels[index].StationLevelData;
+            List<string> labels = GenerateLabelsForStations(levelData);
+
+            List<Transform> usedSpawns = SpawnDeliveries(levelData, labels);
+
+            List<Transform> freeSpawns = GetAllFreeSpawns(usedSpawns);
+
+            SpawnFakeDeliveries(freeSpawns, labels);
+        }
+
+        private List<string> GenerateLabelsForStations(StationLevelData[] stationData)
+        {
+            System.Random random = new System.Random();
+            List<string> labels = new();
+
+            for (int i = 0; i < stationData.Length; i++)
+            {
+                TrainStation targetStation = _allStations[i];
+                string label = string.Empty;
+
+                while (label == string.Empty || labels.Contains(label))
+                {
+                    label
+                        = CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString()
+                        + CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString();
+                }
```

Comment:
Create a method named something like `GetUniqueLabel()` to make this more DRY since you also you this at the fake packages spawning.